### ISSUE TITLE
[Usage] Stuck on Gradio

### ISSUE THREAD

**Issue by chrisk414** [haotian-liu/LLaVA#361]
### Describe the issue

I've installed Llava on Windows (without DeepSpeed and installed Windows version of BitSandBytes), and using all default settings. I'm on GTX4090
The model (llava-llama-2-13b-chat-lightning-preview --load-4bit, on localhost) is loaded correctly and everything seems to work fine without errors, except when I start Web.

Log: 
```
call conda activate llava 

start /b python -m llava.serve.controller --host 0.0.0.0  --port 10000 

start /b python -m llava.serve.gradio_web_server --controller http://localhost:10000 --model-list-mode reload 

start /b python -m llava.serve.model_worker --host 0.0.0.0  --controller http://localhost:10000 --port 40000 --worker http://localhost:40000 --model-path models\llava-llama-2-13b-chat-lightning-preview\ --load-4bit 

(llava) D:\LLaVA>bin D:\anaconda3\envs\llava\lib\site-packages\bitsandbytes\libbitsandbytes_cuda117.dll
bin D:\anaconda3\envs\llava\lib\site-packages\bitsandbytes\libbitsandbytes_cuda117.dll
bin D:\anaconda3\envs\llava\lib\site-packages\bitsandbytes\libbitsandbytes_cuda117.dll
2023-08-09 10:52:49 | INFO | controller | args: Namespace(host='0.0.0.0', port=10000, dispatch_method='shortest_queue')
2023-08-09 10:52:49 | INFO | controller | Init controller
2023-08-09 10:52:49 | INFO | model_worker | args: Namespace(host='0.0.0.0', port=40000, worker_address='http://localhost:40000', controller_address='http://localhost:10000', model_path='models\\llava-llama-2-13b-chat-lightning-preview\\', model_base=None, model_name=None, multi_modal=False, limit_model_concurrency=5, stream_interval=1, no_register=False, load_8bit=False, load_4bit=True)
2023-08-09 10:52:49 | INFO | model_worker | Loading the model models\llava-llama-2-13b-chat-lightning-preview\ on worker 8f446a ...
You are using the legacy behaviour of the <class 'transformers.models.llama.tokenization_llama.LlamaTokenizer'>. This means that tokens that come after special tokens will not be properly handled. We recommend you to read the related pull request available at https://github.com/huggingface/transformers/pull/24565
2023-08-09 10:52:49 | ERROR | stderr | INFO:     Started server process [1952]
2023-08-09 10:52:49 | ERROR | stderr | INFO:     Waiting for application startup.
2023-08-09 10:52:49 | ERROR | stderr | INFO:     Application startup complete.
2023-08-09 10:52:49 | ERROR | stderr | INFO:     Uvicorn running on http://0.0.0.0:10000 (Press CTRL+C to quit)
Loading checkpoint shards:   0%|                                                                                                                                                                             | 0/3 [00:00<?, ?it/s]
2023-08-09 10:53:05 | INFO | gradio_web_server | args: Namespace(host='0.0.0.0', port=None, controller_url='http://localhost:10000', concurrency_count=8, model_list_mode='reload', share=False, moderate=False, embed=False)
2023-08-09 10:53:07 | INFO | stdout | INFO:     127.0.0.1:3866 - "POST /refresh_all_workers HTTP/1.1" 200 OK
2023-08-09 10:53:09 | INFO | stdout | INFO:     127.0.0.1:3868 - "POST /list_models HTTP/1.1" 200 OK
2023-08-09 10:53:09 | INFO | gradio_web_server | Models: []
2023-08-09 10:53:09 | INFO | gradio_web_server | Namespace(host='0.0.0.0', port=None, controller_url='http://localhost:10000', concurrency_count=8, model_list_mode='reload', share=False, moderate=False, embed=False)
Loading checkpoint shards:  33%|███████████████████████████████████████████████████████                                                                                                              | 1/3 [00:22<00:45, 22.86s/it]
2023-08-09 10:53:14 | INFO | stdout | Running on local URL:  http://0.0.0.0:7860
2023-08-09 10:53:14 | INFO | stdout | 
2023-08-09 10:53:14 | INFO | stdout | To create a public link, set `share=True` in `launch()`.
Loading checkpoint shards:  67%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████                                                       | 2/3 [00:43<00:21, 21.43s/it]
Loading checkpoint shards: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 3/3 [00:56<00:00, 17.53s/it]
Loading checkpoint shards: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 3/3 [00:56<00:00, 18.73s/it]
2023-08-09 10:53:47 | ERROR | stderr | 
2023-08-09 10:53:50 | INFO | model_worker | Register to controller
2023-08-09 10:53:52 | INFO | controller | Register a new worker: http://localhost:40000
2023-08-09 10:53:52 | INFO | controller | Register done: http://localhost:40000, {'model_names': ['models\\llava-llama-2-13b-chat-lightning-preview\\'], 'speed': 1, 'queue_length': 0}
2023-08-09 10:53:52 | INFO | stdout | INFO:     127.0.0.1:9716 - "POST /register_worker HTTP/1.1" 200 OK
2023-08-09 10:53:52 | ERROR | stderr | INFO:     Started server process [6508]
2023-08-09 10:53:52 | ERROR | stderr | INFO:     Waiting for application startup.
2023-08-09 10:53:52 | ERROR | stderr | INFO:     Application startup complete.
2023-08-09 10:53:52 | ERROR | stderr | INFO:     Uvicorn running on http://0.0.0.0:40000 (Press CTRL+C to quit)
2023-08-09 10:54:07 | INFO | model_worker | Send heart beat. Models: ['models\\llava-llama-2-13b-chat-lightning-preview\\']. Semaphore: None. global_counter: 0
2023-08-09 10:54:09 | INFO | controller | Receive heart beat. http://localhost:40000
2023-08-09 10:54:09 | INFO | stdout | INFO:     127.0.0.1:9757 - "POST /receive_heart_beat HTTP/1.1" 200 OK
2023-08-09 10:54:24 | INFO | model_worker | Send heart beat. Models: ['models\\llava-llama-2-13b-chat-lightning-preview\\']. Semaphore: None. global_counter: 0
2023-08-09 10:54:26 | INFO | controller | Receive heart beat. http://localhost:40000
2023-08-09 10:54:26 | INFO | stdout | INFO:     127.0.0.1:9787 - "POST /receive_heart_beat HTTP/1.1" 200 OK
2023-08-09 10:54:41 | INFO | model_worker | Send heart beat. Models: ['models\\llava-llama-2-13b-chat-lightning-preview\\']. Semaphore: None. global_counter: 0
2023-08-09 10:54:43 | INFO | controller | Receive heart beat. http://localhost:40000
2023-08-09 10:54:43 | INFO | stdout | INFO:     127.0.0.1:9815 - "POST /receive_heart_beat HTTP/1.1" 200 OK
2023-08-09 10:54:58 | INFO | model_worker | Send heart beat. Models: ['models\\llava-llama-2-13b-chat-lightning-preview\\']. Semaphore: None. global_counter: 0
```

However, the web is stuck on "queue" like the image below. I didn't do anything.
When I refresh the web, I get console messages like the below.
Any ideas? Many thanks.

Log: 
```
2023-08-09 10:56:20 | ERROR | asyncio | Task exception was never retrieved
future: <Task finished name='jzg2326scs_11' coro=<Queue.process_events() done, defined at D:\anaconda3\envs\llava\lib\site-packages\gradio\queueing.py:342> exception=1 validation error for PredictBody
event_id
  Field required [type=missing, input_value={'data': [], 'event_data'...ion_hash': 'jzg2326scs'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.1/v/missing>
Traceback (most recent call last):
  File "D:\anaconda3\envs\llava\lib\site-packages\gradio\queueing.py", line 346, in process_events
    client_awake = await self.gather_event_data(event)
  File "D:\anaconda3\envs\llava\lib\site-packages\gradio\queueing.py", line 219, in gather_event_data
    data, client_awake = await self.get_message(event, timeout=receive_timeout)
  File "D:\anaconda3\envs\llava\lib\site-packages\gradio\queueing.py", line 448, in get_message
    return PredictBody(**data), True
  File "D:\anaconda3\envs\llava\lib\site-packages\pydantic\main.py", line 159, in __init__
    __pydantic_self__.__pydantic_validator__.validate_python(data, self_instance=__pydantic_self__)
pydantic_core._pydantic_core.ValidationError: 1 validation error for PredictBody
event_id
  Field required [type=missing, input_value={'data': [], 'event_data'...ion_hash': 'jzg2326scs'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.1/v/missing
```

Screenshots:
![image](https://github.com/haotian-liu/LLaVA/assets/752887/115dd23b-45ae-4c77-b983-b87ce52c2bf4)

**Comment by blankxxc** [comment]
I have this question too .Are you solve?

**Comment by natlamir** [comment]
Downgrading pyndantic fixed it for me:
```
pip install pydantic==1.10.9
```

**Comment by haotian-liu** [comment]
Please check the latest doc for Windows. I tested on my Windows 11 PC and it works now for 16-bit inference. Quantization will be supported later.

https://github.com/haotian-liu/LLaVA/blob/main/docs/Windows.md

**Comment by Coldmooon** [comment]
I encountered the same issue. I solve this by
```
sudo rm /etc/X11/xorg.conf
```

and then `sudo reboot`

My server is headless. So I install a virtual display and set the `xorg.conf` accordingly. When I encountered this issue, I also found the virtual display is down. Then I connected a real display to my server and found the destop of ubuntu system is also down. So I simply delete the `xorg.conf` file and disable the virtual display. The issue disappeared.
